### PR TITLE
Move logic from connect-api to connect

### PR DESCRIPTION
**Pull request by valentinludu**

### CODE REVIEW

File: @connect-shared/components/common/MultiTextInputField.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ export function MultiTextInputField<T extends FieldValues>({
               placeholder={placeholder}
               error={!!fieldState.error}
               {..._field}
+              value={_field.value || ''}
```
Comment (mattcasey) [inline @connect-shared/components/common/MultiTextInputField.tsx:48]: I think u can remove this - i believe i fixed it by providing defaultValue

File: @connect-shared/lib/session/logoutAction.ts [hunk 2]
```diff
@@ -4,11 +4,11 @@ import { log } from '@charmverse/core/log';
 
 import { authActionClient } from '../actions/actionClient';
 
-export const logoutAction = authActionClient.metadata({ actionName: 'login' }).action(async ({ ctx }) => {
+export const logoutAction = authActionClient.metadata({ actionName: 'logout' }).action(async ({ ctx }) => {
```
Comment (mattcasey) [inline @connect-shared/lib/session/logoutAction.ts:7]: thanks for fixing!

File: apps/connect/app/api/aws/upload-token/route.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import { GetFederationTokenCommand, STSClient } from '@aws-sdk/client-sts';
```
Comment (mattcasey) [inline apps/connect/app/api/aws/upload-token/route.ts:1]: Please dont duplicate code - we have this already written inside sunnyawards, so it should be put into connect-shared

File: apps/connect/components/projects/components/ProjectForm.tsx [hunk 4]
```diff
@@ -127,6 +128,7 @@ export function ProjectForm({
                   aria-labelledby='project-twitter'
                   error={!!fieldState.error}
                   {...field}
+                  value={field.value || ''}
```
Comment (mattcasey) [inline apps/connect/components/projects/components/ProjectForm.tsx:131]: this looks like a hack, i am not sure what problem you are solving. maybe we need to be providing default values in the useForm or provide a defaultValue instead?

File: apps/sunnyawards/components/common/Header/Header.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ export function Header({ user }: { user: LoggedInUser | null }) {
 
   const { execute: logoutUser } = useAction(logoutAction, {
     onSuccess: async () => {
-      await revalidatePathAction();
+      revalidatePathAction();
```
Comment (mattcasey) [inline apps/sunnyawards/components/common/Header/Header.tsx:31]: why should we not await? this function returns a promise?

**Review by mattcasey — CHANGES_REQUESTED** [review]
Sunny Awards and Connect should share code that is the same. When I created Connect, we said that it was not worth me going back and de-duping at the time, and agreed to do it as-we-go. Please at least remove the new duplicate code for uploading images. And ideally, now that we are only using server actions in both it would be easy to re-use some other components like SearchFarcasterUser.

File: @connect-shared/components/common/MultiTextInputField.tsx (hunk 1 above)
Comment (valentinludu) [reply @connect-shared/components/common/MultiTextInputField.tsx:48]: defaultValue makes a component to be uncontrolled and then, when the state changes it becomes controlled and the browser throws an error. Usually React recommends to create a component controlled or uncontrolled.

Screenshot 2024-08-13 at 10 11 20

Comment (valentinludu) [reply @connect-shared/components/common/MultiTextInputField.tsx:48]: Another way would be to set default values to useForm hook.

File: apps/connect/app/api/aws/upload-token/route.ts (hunk 3 above)
Comment (valentinludu) [reply apps/connect/app/api/aws/upload-token/route.ts:1]: Agree. Moved it in connect-shared folder!

File: apps/connect/components/projects/components/ProjectForm.tsx (hunk 4 above)
Comment (valentinludu) [reply apps/connect/components/projects/components/ProjectForm.tsx:131]: I commented above.

File: apps/sunnyawards/components/common/Header/Header.tsx (hunk 5 above)
Comment (valentinludu) [reply apps/sunnyawards/components/common/Header/Header.tsx:31]: It makes the change of the page faster. We can discuss more in our call.

**Review by mattcasey — APPROVED** [review]

**Pull request merged**